### ISSUE TITLE
Generator copyWith

### ISSUE THREAD

**Issue by saddig** [isar/isar#489]
In `3.0.0-dev.7` the generator fails to generate collections for freezed entities due to the generated copyWith methode.

[SEVERE] isar_generator:isar_generator on lib/src/data/models/foo.dart:

Unsupported type. Please use a TypeConverter or annotate the propery with `@ignore`.

```
@JsonKey(ignore: true)
$FooCopyWith<Foo> get copyWith => throw _privateConstructorUsedError;
```  
  

Isar's `@ignore` annotation should also ignore properties annotated with the `@JsonKey(ignore: true)` annotation.

**Comment by Ludonope** [comment]
Still present in `3.0.0-dev.8`.

**Comment by pMertDogan** [comment]
Hi,

Looks like same issue 

![image](https://user-images.githubusercontent.com/23584565/184556545-1ff0414f-224b-4b71-b5a7-fcd1f72ba006.png)

**Comment by abdhoms** [comment]
Any fix or workaround for this?

**Comment by leisim** [comment]
@abdhoms No because I don't really know how to fix it. Just ignoring all fields that json ignores feels really weird and unexpected. Isar has nothing to do with json and there are many usecases where fields should be ignored for json serialization but not for Isar.

If anyone knows how we could solve this please let me know.

**Comment by leisim** [comment]
Maybe a `ignore` property of `@Collection` where a list of fields can be provided?

Like `@Collection(ignore: ['copyWith'])`

**Comment by leisim** [comment]
This has been implemented in `3.0.0-dev.14`
